arviz/data/inference_data.py

```diff
@@ -241,6 +241,61 @@ def to_netcdf(self, filename, compress=True, groups=None):
             empty_netcdf_file.close()
         return filename
 
+    def to_dict(self, groups=None):
+        """Convert InferenceData to a dictionary following xarray naming
+        conventions.
+
+        Parameters
+        ----------
+        groups : list, optional
+            Write only these groups to netcdf file.
+
+        Returns
+        -------
+        dict
+            A dictionary containing all groups of InferenceData object.
+            When `data=False` return just the schema.
+        """
+        ret = {}
+        ret.setdefault("coords", dict())
+        ret.setdefault("dims", dict())
+        ret.setdefault("pred_dims", dict())
+        attrs = None
+        if self._groups_all:  # check's whether a group is present or not.
+            if groups is None:
+                groups = self._groups_all
+            else:
+                groups = [group for group in self._groups_all if group in groups]
+
+            for group in groups:
+                xr_data = getattr(self, group)
+                ds = xr_data.data_vars
+                data = {}
+                for key, value in ds.items():
+                    data[key] = value.values
+                    dims = []
+                    for k_, v_ in value.coords.items():
```

Comment:
I would rename many of these variables to make the code easier to understand. 
* `xr_data` -> `ds`/`dataset` (or use group and modify group in loop to group_name)
  * xarray datasets have a [dictionary interface](http://xarray.pydata.org/en/stable/api.html#dictionary-interface) so there is not need to do `.data_vars`, looping over `xarray.Dataset.items()` does the same.
* `key` -> `var_name`
* `value` -> `da`/`dataarray`
* `k_` -> `dim`/`coord_name`
* `v_` -> `coord_values`